activation/activation.go

```diff
@@ -652,24 +652,12 @@ func (b *Builder) createAtx(
 		return nil, fmt.Errorf("%w: atx publish epoch has passed during nipost construction", ErrATXChallengeExpired)
 	}
 
-	var nonce *types.VRFPostIndex
 	var atxNodeID *types.NodeID
-	switch {
-	case challenge.PrevATXID == types.EmptyATXID:
+	if challenge.PrevATXID == types.EmptyATXID {
 		atxNodeID = new(types.NodeID)
 		*atxNodeID = sig.NodeID()
-		nonce = &nipostState.VRFNonce
-	default:
-		oldNonce, err := atxs.VRFNonce(b.cdb, sig.NodeID(), challenge.PublishEpoch)
-		if err != nil {
-			b.log.Warn("failed to get VRF nonce for ATX", zap.Error(err), log.ZShortStringer("smesherID", sig.NodeID()))
-			break
-		}
-		if nipostState.VRFNonce != oldNonce {
-			nonce = &nipostState.VRFNonce
-		}
 	}
-
+	nonce := &nipostState.VRFNonce
```

Comment:
Why always include the VRFNonce? It only needs to be included in the initial ATX and in case it changes (i.e. when a user resizes their PoST).

Reply:
Always including nonce in ATX substantially speeds up lots of queries, see this comment https://github.com/spacemeshos/go-spacemesh/pull/5791#discussion_r1545473980
and the issue https://github.com/spacemeshos/go-spacemesh/issues/5701

Reply:
No, sorry. There is no point in broadcasting values to other peers that they already know about. I don't have an issue with storing the VRFNonce for every ATX in the DB to speed up DB queries, but I object to including it in every broadcast ATX.

Reply:
Reverted this change